### PR TITLE
fixing panic on peer list due to empty map

### PR DESCRIPTION
**Pull request by akoserwal**
When no device is registered as peers, result is empty

related to https://github.com/nexodus-io/nexodus/issues/1165

When only single device is registered

```
nexctl nexd peers list
panic: reflect: call of reflect.Value.FieldByName on map Value

goroutine 1 [running]:
reflect.flag.mustBe(...)
/usr/local/go/src/reflect/value.go:223
reflect.Value.FieldByName({0x50d8a0?, 0x400000e868?, 0x0?}, {0x5b9d86?, 0x1?})
/usr/local/go/src/reflect/value.go:1345 +0x13c
main.showOutput(0x4000368260?, {0x4000311180, 0x3, 0x4cc980?}, {0x50d8a0?, 0x4000169dd0})
/src/cmd/nexctl/main.go:612 +0x7a4
main.cmdListPeers(0x18?, {0x5b54b1?, 0x6?})
/src/cmd/nexctl/peers.go:73 +0x118
main.init.1.func8(0x4000358000?)
/src/cmd/nexctl/local_unix.go:212 +0x3c
[[github.com/urfave/cli/v2.(*Command](http://github.com/urfave/cli/v2.(*Command)](http://[github.com/urfave/cli/v2.(*Command](http://github.com/urfave/cli/v2.(*Command))).Run(0x4000358000, 0x40003209c0, {0x40003668e0, 0x1, 0x1})
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.3/command.go:274 +0x73c
[[github.com/urfave/cli/v2.(*Command](http://github.com/urfave/cli/v2.(*Command)](http://[github.com/urfave/cli/v2.(*Command](http://github.com/urfave/cli/v2.(*Command))).Run(0x4000157e40, 0x4000320900, {0x400003b880, 0x2, 0x2})
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.3/command.go:267 +0x948
[github.com/urfave/cli/v2.(*Command](http://github.com/urfave/cli/v2.(*Command)).Run(0x4000156b00, 0x4000320780, {0x40001699e0, 0x3, 0x3})
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.3/command.go:267 +0x948
[github.com/urfave/cli/v2.(*Command](http://github.com/urfave/cli/v2.(*Command)).Run(0x400035c6e0, 0x4000320500, {0x4000030080, 0x4, 0x4})
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.3/command.go:267 +0x948
[[github.com/urfave/cli/v2.(*App](http://github.com/urfave/cli/v2.(*App)](http://[github.com/urfave/cli/v2.(*App](http://github.com/urfave/cli/v2.(*App))).RunContext(0x400011a5a0, {0x69a030?, 0x4000036190}, {0x4000030080, 0x4, 0x4})
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.3/app.go:332 +0x568
[github.com/urfave/cli/v2.(*App](http://github.com/urfave/cli/v2.(*App)).Run(...)
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.3/app.go:309
main.main()
/src/cmd/nexctl/main.go:474 +0x1e64
```

### CODE REVIEW

**Review by russellb — CHANGES_REQUESTED** [review]

File: cmd/nexctl/peers.go [hunk 1]
```diff
@@ -69,7 +69,11 @@ func cmdListPeers(cCtx *cli.Context, encodeOut string) error {
 	if err != nil {
 		return fmt.Errorf("Failed to marshall peer results: %w\n", err)
 	}
+	if len(peers) == 0 {
+		fmt.Printf("No peers found %v\n", peers)
+	} else {
+		showOutput(cCtx, peerTableFields(cCtx), peers)
+	}
```
[the remark this answers is not in the document]
Comment (akoserwal) [reply cmd/nexctl/peers.go:76]: I agree, I will work on it

File: cmd/nexctl/main.go [hunk 2]
```diff
@@ -591,34 +591,37 @@ func showOutput(cCtx *cli.Context, fields []TableField, result any) {
...
-			var line []string
-			for _, field := range fields {
-				if field.Formatter != nil {
-					line = append(line, field.Formatter(itemValue.Interface()))
-				} else if field.Field != "" {
-					// Deref the items points.
-					for itemValue.Type().Kind() == reflect.Pointer {
-						itemValue = itemValue.Elem()
+		if !itemsValue.IsNil() {
+			fmt.Printf("resource not found: %v\n", itemsValue)
```
Comment (russellb) [inline cmd/nexctl/main.go:596]: I think I would prefer the header row and then no output afterwards

Comment (russellb) [reply cmd/nexctl/main.go:596]: s/fmt.Printf(.*)/break/

and then you don't need an `else` block at all

**Comment by chirino** [comment]
Would have been nice if the e2e tests included this edge case to make sure this does not regress in the future.  It could be as easy as doing a `device list` before a nexd agent gets started in any of the e2e tests.

**Comment by akoserwal** [comment]
> Would have been nice if the e2e tests included this edge case to make sure this does not regress in the future. It could be as easy as doing a `device list` before a nexd agent gets started in any of the e2e tests.

On running
`nexctl nexd peers list` before a nexd agent gets started in any of the e2e tests.
Error
`Failed to connect to nexd: dial unix nexd.sock: connect: no such file or directory`

This will work
`statOut, err := helper.containerExec(ctx, ctr, []string{"/bin/nexctl", "nexd", "peers", "list"})`

^ will work post merging the PR. Test container nexctl points to old version.

**Comment by chirino** [comment]
> ^ will work post merging the PR. Test container nexctl points to old version.

I sure that `make e2e` on this PR would test your fix, so you could add that to the e2e test.

File: integration-tests/integration_test.go [hunk 3]
```diff
@@ -894,6 +894,10 @@ func TestNexctl(t *testing.T) {
 		}
 	}
 
+	// verify nexd peers list
+	err = helper.peerListNexdDevices(ctx, node1)
```
Comment (chirino) [inline integration-tests/integration_test.go:898]: Can't we do this before even nexd is started up?  Like around like 825?

**Review by chirino — APPROVED** [review]
Lgtm
